controllers/repository_controller.go

```diff
@@ -143,6 +144,12 @@ func (r *RepositoryReconciler) checkInitial(ctx context.Context, logger logr.Log
 
 	if v, ok := instanceDeepCopy.Labels[corev1alpha1.RepositoryTypeLabel]; !ok || v != instanceDeepCopy.Spec.RepositoryType {
 		instanceDeepCopy.Labels[corev1alpha1.RepositoryTypeLabel] = instanceDeepCopy.Spec.RepositoryType
+		if !instanceDeepCopy.IsOCI() {
```

Comment:
You can place this line
`if !instanceDeepCopy.IsOCI() {`

in your `getRepoType` too.